### DISCUSSION TITLE
Incorrectly generated query with many to one primary key

### DISCUSSION THREAD

**Discussion by alexrequelme** [mikro-orm/mikro-orm#3707] [Q&A]
**Describe the bug**
The sql generated by mikro-orm is wrong when querying a table with many to one relationships. I want to check the winners of a product that has the pks of the participating table, but when filtering by user it generates the comparison by product_id

**Stack trace**
select "g0".* from "winners" as "g0" where "g0"."product_id" = 1 - operator does not exist: text = integer

**To Reproduce**

```javascript
@Entity()
export class ProductParticipants {
  @ManyToOne({
    primary: true,
    fieldNames: ['combination_id', 'product_id'],
    hidden: true,
  })
  productCombination: productCombination;

  @ManyToOne({ primary: true, hidden: true })
  user: User;

  constructor(productParticipants : Partial<ProductParticipants>) {
    Object.assign(this, ProductParticipants );
  }
}
```
```javascript
@Entity()
export class ProductWinners {
  @ManyToOne({
    primary: true,
    fieldNames: ['combination_id', 'product_id', 'user_id'],
    hidden: true,
  })
  productParticipants: productParticipants;

  @Property()
  isClaimed = false;

  constructor(productWinners: Partial<ProductWinners>) {
    Object.assign(this, productWinners);
  }
}
```
```javascript
return this.winnersRepository.find({ productParticipants : { user } });
```

**Expected behavior**
select "g0".* from "winners" as "g0" where "g0"."user_id" = 1 - operator does not exist: text = integer

**Additional context**

**Versions**

| Dependency | Version |
| - | - |
| node | v16.16.0 |
| typescript | 4.7.4 |
| mikro-orm | 5.4.2 |
| your-driver | postgresql |

**Comment by B4nan** [comment]
> mikro-orm | 5.4.2

please verify with latest version first

**Reply by alexrequelme** [reply]
I already tried it with the latest version and it gives me the same problem. I also realized that when uploading from 5.4.2 to 5.5.2, I found a problem and I don't know if it's the expected behavior, but the way I saved an entity in my table stopped working, for example if I want create a product:

```javascript
this.productRepository.create({
  .....,
  owner: userId // this works,
  store: {
    storeId,
    countryId, // both pks, dont work
  }
})
```

Now it asks me for the rest of the fields to create the entity, when before it only asked me for the composite pks.

**Reply by B4nan** [reply]
Try this:

```ts
this.productRepository.create({
  .....,
  owner: userId // this works,
  store: [storeId, countryId]
})
```

That is how a composite PK is represented in a DTO. Will take a look at that locally, if the object form worked in 5.4.2, it should keep working.

**Reply by alexrequelme** [reply]
Thanks, thats works but now:

```javascript
@OneToMany(() =>ProductParticipants, (value) => value.productCombination)
  participants = new Collection<ProductParticipants>(this);
```

Value for ProductParticipants.user is required, 'undefined' found, this also did not fail in version 5.4.2, it is supposed to be created with participants [ ] empty at the beginning

**Reply by alexrequelme** [reply]
It seems that it is validating extra fields that should not be validated

**Reply by B4nan** [reply]
I really cant help you more unless you provide complete repro. I have no idea what you are doing, on the other hand I have 1600+ passing tests cases in the ORM test suite, as well as several example apps that are tested via CI agains the latest versions...

**Reply by alexrequelme** [reply]
Ok I'm going to upload a repo to show with version 5.5.2, although the strange thing is that with 5.5.0 it doesn't give me an error.

**Reply by B4nan** [reply]
The version is not that relevant if I can easily change it myself, feel free to use 5.4.2 if the repro will be complete.

**Reply by B4nan** [reply]
> Value for ProductParticipants.user is required, 'undefined' found, this also did not fail in version 5.4.2, it is supposed to be created with participants [ ] empty at the beginning

I can reproduce this, and it behaves exactly the same in 5.4.2, same with 5.3.0 - in fact it did behave this way since 5.0 if I am not mistaken.

Note that I am testing the `em.create()` call, I guess in your case it might have been triggered internally. Again, without repro, I have no idea what's happening and why.

**Comment by B4nan** [comment]
If that won't work, I will need complete reproduction - at least complete entity definitions + ORM config.
